test/functional/rpc_blockchain.py

```diff
@@ -99,7 +106,8 @@ def _test_getblockchaininfo(self):
         ]
         res = self.nodes[0].getblockchaininfo()
 
-        assert isinstance(res['time'], int)
+        assert_equal(res['time'], TIME_RANGE_END - TIME_RANGE_STEP)
+        assert_equal(res['mediantime'], TIME_RANGE_MTP)
```

Comment:
It would be interesting if the TIME_RANGE_STEP varied quasi-randomly by +/- 3 minutes. Wouldn't having a variation be a better test? Just a thought. :)

Reply:
The `getnetworkhashps` test would need to be adapted to it, otherwise seems doable. Feel free to propose it as a follow-up!